plugin.rb

```diff
@@ -229,6 +237,11 @@ def fetch_user_details(token, id)
         json_walk(result, user_json, :email)
         json_walk(result, user_json, :email_verified)
         json_walk(result, user_json, :avatar)
+
+        DiscoursePluginRegistry.custom_user_details.each do |detail|
+          prop = detail.split('.').last.to_sym
```

Comment:
I think this might risk a clash? e.g. if `prop` was something like `username`, it would override the oauth2 plugin's own username value.

Maybe we could use the whole `detail`, and it could be namespaced like

```
prop = "extra:#{detail}"
```
(I don't *think* there is any need for it to be a symbol, so a string with special characters should work)